### ISSUE TITLE
Access true label

### ISSUE THREAD

**Issue by qz28** [apache/incubator-mxnet#5351]
I am a beginner and try to use a pre-trained resnet. I have a problem of accessing the ground truth label. I looked at some tutorial and still got quite confused:

sym, arg_params, aux_params = mx.model.load_checkpoint('resnet-152', 0)
all_layers = sym.get_internals()
net = all_layers['flatten0'+'_output']
net = mx.symbol.FullyConnected(data=net, num_hidden=num_classes, name='fc1')
net = mx.symbol.SoftmaxOutput(data=net, name='softmax')

This softmaxoutput layer doesn't seem to take the true label as a argument (which is described as a non-optional parameter on the website). How the loss is calculated when label is not passed to the softmax output layer? I looked at the input list of symbol sym(the pretrained resnet model) and found softmax_label is in there. I want to scale the loss contribution of each sample based on their true labels. How can I access the labels under this circumstances ? And How can I access the loss it seems the loss is computed in the softmaxoutput layer but not outputted.?

For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System:

Compiler:

Package used (Python/R/Scala/Julia):

MXNet version:

Or if installed from source:

MXNet commit hash (`git rev-parse HEAD`):

If you are using python package, please provide

Python version and distribution:

If you are using R package, please provide

R `sessionInfo()`:

## Error Message:
Please paste the full error message, including stack trace.

## Minimum reproducible example
if you are using your own code, please provide a short script that reproduces the error.

## Steps to reproduce
or if you are running standard examples, please provide the commands you have run that lead to the error.

1.
2.
3.

## What have you tried to solve it?

1.
2.
3.

**Comment by saicoco** [comment]
The method `provide_label` in DataIter returns `[('softmax_label, (batch_size,))]`, and mxnet get input or  output of symbol by their names.So, if you dont diy your label's name, you can use the default format; or you should set param label in `SoftmaxOutput`.For example as follows:  

```python
label = mx.sym.Variable('my_label')
...
softmax = mx.sym/SoftmaxOutput(data=data, label=label)
```  

And then, the `provide_label` in your dataIter should provide `[('my_label', (batch_size, ))]`

**Comment by qz28** [comment]
@saicoco 
Thanks for your  
I tried 
label = mx.sym.Variable( 'softmax_label')
but this gave me a duplicate name error. And I used list_arguments() to look the input list of my modified pre-trained model, it has two variables named 'softmax_label'.  The first one is the original and the second should be the one I introduced. I guess it doesn't allow me to use mx.sym.Variable( 'softmax_label') to access the ground truth label? Is this a normal situation?

**Comment by saicoco** [comment]
There is not label-variable in pre-trained model , here is my example about vgg:

```python
def get_model(checkpoints):
    sym, arg_params, aux_params = mx.model.load_checkpoint(checkpoints, 0)
    return sym, arg_params, aux_params
    
def get_finetune_model(sym, params, num_classes, layer_name='fc7'):
    label = mx.sym.Variable('softmax_label')
    all_layers = sym.get_internals()
    net = all_layers[layer_name + '_output']
    net = mx.sym.FullyConnected(data=net, num_hidden=num_classes, name='newf8')
    net = mx.sym.SoftmaxOutput(data=net, label=label, name='softmax')
    new_args = dict(
        {k:params[k] for k in params.keys() if 'newf8' is not k}
    )
    return (net, new_args)

if __name__=='__main__':
    sym, args_params, aux_params = get_model('/home/sai/LIB/mxnet/example/finetune/vgg16')
    print 'pe_trained:{}'.format(sym.list_arguments())
    net, new_args = get_finetune_model(sym, args_params, 6)
    print 'modefiy:{}'.format(net.list_arguments())
```  

and result:  

```
pe_trained:['data', 'conv1_1_weight', 'conv1_1_bias', 'conv1_2_weight', 'conv1_2_bias', 'conv2_1_weight', 'conv2_1_bias', 'conv2_2_weight', 'conv2_2_bias', 'conv3_1_weight', 'conv3_1_bias', 'conv3_2_weight', 'conv3_2_bias', 'conv3_3_weight', 'conv3_3_bias', 'conv4_1_weight', 'conv4_1_bias', 'conv4_2_weight', 'conv4_2_bias', 'conv4_3_weight', 'conv4_3_bias', 'conv5_1_weight', 'conv5_1_bias', 'conv5_2_weight', 'conv5_2_bias', 'conv5_3_weight', 'conv5_3_bias', 'fc6_weight', 'fc6_bias', 'fc7_weight', 'fc7_bias', 'fc8_weight', 'fc8_bias', 'prob_label']
modefiy:['data', 'conv1_1_weight', 'conv1_1_bias', 'conv1_2_weight', 'conv1_2_bias', 'conv2_1_weight', 'conv2_1_bias', 'conv2_2_weight', 'conv2_2_bias', 'conv3_1_weight', 'conv3_1_bias', 'conv3_2_weight', 'conv3_2_bias', 'conv3_3_weight', 'conv3_3_bias', 'conv4_1_weight', 'conv4_1_bias', 'conv4_2_weight', 'conv4_2_bias', 'conv4_3_weight', 'conv4_3_bias', 'conv5_1_weight', 'conv5_1_bias', 'conv5_2_weight', 'conv5_2_bias', 'conv5_3_weight', 'conv5_3_bias', 'fc6_weight', 'fc6_bias', 'fc7_weight', 'fc7_bias', 'newf8_weight', 'newf8_bias', 'softmax_label']
```  

just load it and then to define youself model.

**Comment by qz28** [comment]
@saicoco Thank you! That's very helpful!

By the way, do you know how to access the loss? It seems like the loss is computed within the softmax ouput layer. Is there a symbol for the loss which I can adapt into a custom loss?

**Comment by saicoco** [comment]
`mx.sym.MakeLoss`

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
